scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -2115,7 +2115,7 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
 
   def macroSplicedIdent(): Term = autoPos {
     token match {
-      case Ident(value) =>
+      case Ident(value) if value.size > 1 =>
```

Comment:
perhaps length instead of size?

also, what if the value doesn't start with `$` that we're trying to strip?

Reply:
> > perhaps length instead of size?
> 
> The definition of size is `@`inline` def size: Int = s.length` so it looks to result in the same behavior.

what is the benefit of using size instead of the more explicit length? if i remember correctly, size carries the meaning that it might potentially need to iterate over the entire sequence while length is more likely to be O(1).

> 
> > also, what if the value doesn't start with $ that we're trying to strip?
> 
> the current behaviour is that `macroSplicedIdent` is only invoked if there is `$`. So currently there is no way that it doesn't start with $

I understand that. but what if the behavior changes, and the method is invoked without this guarantee? especially since this is a public method.

Reply:
Ok, changed to `length`, I just thought that there is no real difference.

> I understand that. but what if the behavior changes, and the method is invoked without this guarantee? especially since this is a public method.

Honestly, I don't know what should happen if there is no `$`, that shouldn't happen unless someone breaks the parser code.  Added another check just in case.